specification/botservice/resource-manager/Microsoft.BotService/stable/2020-06-02/botservice.json

```diff
@@ -0,0 +1,2550 @@
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/Channel",
+          "description": "The set of properties specific to bot channel resource"
+        }
+      }
+    },
+    "Channel": {
+      "type": "object",
+      "description": "Channel definition",
+      "discriminator": "channelName",
+      "properties": {
+        "channelName": {
+          "type": "string",
+          "description": "The channel name"
+        }
+      },
+      "required": [
+        "channelName"
+      ]
+    },
+    "AlexaChannel": {
+      "type": "object",
+      "x-ms-discriminator-value": "AlexaChannel",
+      "description": "Alexa channel definition",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Channel"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AlexaChannelProperties",
+          "description": "The set of properties specific to Alexa channel resource"
+        }
+      }
+    },
+    "AlexaChannelProperties": {
+      "properties": {
+        "alexaSkillId": {
+          "type": "string",
+          "description": "The Alexa skill Id"
+        },
+        "urlFragment": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Url fragment used in part of the Uri configured in Alexa"
+        },
+        "serviceEndpointUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Full Uri used to configured the skill in Alexa"
+        },
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Whether this channel is enabled for the bot"
+        }
+      },
+      "description": "The parameters to provide for the Alexa channel.",
+      "required": [
+        "alexaSkillId",
+        "isEnabled"
+      ]
+    },
+    "FacebookChannel": {
+      "type": "object",
+      "description": "Facebook channel definition",
+      "x-ms-discriminator-value": "FacebookChannel",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Channel"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/FacebookChannelProperties",
+          "description": "The set of properties specific to bot facebook channel"
+        }
+      }
+    },
+    "FacebookChannelProperties": {
+      "properties": {
+        "verifyToken": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Verify token. Value only returned through POST to the action Channel List API, otherwise empty."
+        },
+        "pages": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FacebookPage"
+          },
+          "description": "The list of Facebook pages"
+        },
+        "appId": {
+          "type": "string",
+          "description": "Facebook application id"
+        },
+        "appSecret": {
```

Comment:
Is this a secret? If yes, it should be marked with `x-ms-secret: true` extension.

Reply:
I don't see it or other x-ms-secret changes. Did you forget to push?

Reply:
I haven't pushed yet. I am still working on the id and type fixes.

Reply:
np - I'll wait for the commit notification.

Reply:
Looks good now.